src/Entities/Entity.cpp

```diff
@@ -457,10 +457,15 @@ bool cEntity::DoTakeDamage(TakeDamageInfo & a_TDI)
 					case mtSilverfish:
 					{
 						a_TDI.RawDamage += static_cast<int>(ceil(2.5 * BaneOfArthropodsLevel));
-						// TODO: Add slowness effect
+						// The duration of the effect is a random value between 1 and 1.5 seconds at level I,
+						// increasing the max duration by 0.5 seconds each level, up to 3.5s with Bane of Arthropods V.
+						// Ref: https://minecraft.gamepedia.com/Enchanting#Bane_of_Arthropods
+						double MaxDuration = std::min(1 + BaneOfArthropodsLevel * 0.5, 3.5);
+						double Duration = GetRandomProvider().RandReal<double>(1, MaxDuration * 20);
+						Monster->AddEntityEffect(cEntityEffect::effSlowness, static_cast<int>(ceil(Duration)), 4);
```

Comment:
`static_cast<int>(1 + BaneOfArthropodsLevel * 0.5)` will return 1, 2, or 3, and the random will then only choose an exact number of seconds (that will never give, say, 1.5 seconds).

If you multiply by 20 inside the RandInt it should work fine, though:
```
int Duration = 20 + GetRandomProvider().RandInt(0, static_cast<int>(10 * BaneOfArthropodsLevel));
Monster->AddEntityEffect(cEntityEffect::effSlowness, Duration, 4);
```
(pulled the 20 out in front, so there weren't a bunch of 20s everywhere)